superset/connectors/sqla/models.py

```diff
@@ -231,14 +222,20 @@ def is_temporal(self) -> bool:
         """
         if self.is_dttm is not None:
             return self.is_dttm
-        column_spec = self.table.database.db_engine_spec.get_column_spec(self.type)
-        if column_spec is None:
-            return False
-        return column_spec.is_dttm
+        return self.type_generic == GenericDataType.TEMPORAL
+
+    @property
+    def db_engine_spec(self) -> Type[BaseEngineSpec]:
+        return self.table.db_engine_spec
+
+    @property
+    def type_generic(self) -> Optional[utils.GenericDataType]:
```

Comment:
consider renaming to `get_generic_type` gives an head warning that it will do some computation around it

Reply:
This is added to the column payload in the dataset request to complement the existing `type` field, so I think we need to keep it as a property.